numba/core/ir_utils.py

```diff
@@ -2056,20 +2056,29 @@ def resolve_mod(mod):
         return None
 
 
+def enforce_no_dels(func_ir):
+    """
+    Enforce there being no ir.Del nodes in the IR.
+    """
+    strict = True
+    for blk in func_ir.blocks.values():
+        dels = [x for x in blk.find_insts(ir.Del)]
+        if dels:
+            msg = "Illegal IR, del found at: %s" % dels[0]
+            if strict:
+                raise CompilerError(msg, loc=dels[0].loc)
```

Comment:
Can we have a direct test for this function?

Reply:
done in e977137